ngmix/flags.py

```diff
@@ -74,10 +76,14 @@ def get_flags_str(val, name_map=None):
     if name_map is None:
         name_map = NAME_MAP
 
+    if val < 0 or val >= 2**32:
+        import warnings
+        warnings.warn(f"Flag value {val} is out of range [0, 2**32).")
+
     nstrs = []
-    for pow in range(32):
+    for pow in range(31):
         fval = 2**pow
-        if ((val & fval) != 0):
+        if ((np.array(val).astype(np.uint32) & fval) != 0):
```

Comment:
Let's move this cast up to the top and do it once.

Reply:
I want to make sure we don't do too many redundant things. If we limit the loop to `range(31)`, then casting isn't even required. The error was arising from implicit casting of `fval` and not `val`. We should perhaps raise a `ValueError` if `val` is outside the specific range to avoid any underflow errors be done with it. For instance, if `val = np.int32(-5)`, we'd get a completely wrong result with just a seemingly benign warning.

Reply:
My suggestion is to modify your list above to the following:

We should

- limit the loop to 31
- raise an exception if the input value is bigger than 2**32-1 or < 0

Reply:
OK, in that case, having an upper limit doesn't make sense then. We raise an exception if `val` is negative and cast it to `uint64` safely always.

Reply:
Well, what do you know. The code will now work even if `val` is `np.int8` etc.

Reply:
My plan of always casting to `uint64` doesn't work for `numpy<2`. However, given the range of values that `fval` takes, which is hard coded in this routine,`uint32` should be sufficient. It does not pose any restriction on the exact integer type of `val` .

Reply:
I'd hate to break anything because I didn't consider a corner case, so I added a small unit test as well.